lib/IRGen/GenDecl.cpp

```diff
@@ -640,10 +640,15 @@ emitGlobalList(IRGenModule &IGM, ArrayRef<llvm::WeakTrackingVH> handles,
           IGM.addUsedGlobal(var);
       }
 
+      // Keep classes and categories around for potential ObjC interop.
+      if (IGM.IRGen.Opts.SafeConditionalRuntimeRecords &&
```

Comment:
At some point this was found to be unsafe, though I wasn't around when the issue was discovered. I've thought about removing this restriction a few times, but it seems unsafe to me. IIUC `objc_classes` and `objc_categories` sections are read by the runtime, and so usages would not be detectable statically.